flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -184,46 +168,46 @@ class TableImpl(
   }
 
   override def join(right: Table): Table = {
-    joinInternal(right, None, JoinType.INNER)
+    join(right, None, JoinType.INNER)
   }
 
   override def join(right: Table, joinPredicate: String): Table = {
     join(right, ExpressionParser.parseExpression(joinPredicate))
   }
 
   override def join(right: Table, joinPredicate: Expression): Table = {
-    joinInternal(right, Some(joinPredicate), JoinType.INNER)
+    join(right, Some(joinPredicate), JoinType.INNER)
   }
 
   override def leftOuterJoin(right: Table): Table = {
-    joinInternal(right, None, JoinType.LEFT_OUTER)
+    join(right, None, JoinType.LEFT_OUTER)
   }
 
   override def leftOuterJoin(right: Table, joinPredicate: String): Table = {
     leftOuterJoin(right, ExpressionParser.parseExpression(joinPredicate))
   }
 
   override def leftOuterJoin(right: Table, joinPredicate: Expression): Table = {
-    joinInternal(right, Some(joinPredicate), JoinType.LEFT_OUTER)
+    join(right, Some(joinPredicate), JoinType.LEFT_OUTER)
   }
 
   override def rightOuterJoin(right: Table, joinPredicate: String): Table = {
     rightOuterJoin(right, ExpressionParser.parseExpression(joinPredicate))
   }
 
   override def rightOuterJoin(right: Table, joinPredicate: Expression): Table = {
-    joinInternal(right, Some(joinPredicate), JoinType.RIGHT_OUTER)
+    join(right, Some(joinPredicate), JoinType.RIGHT_OUTER)
   }
 
   override def fullOuterJoin(right: Table, joinPredicate: String): Table = {
     fullOuterJoin(right, ExpressionParser.parseExpression(joinPredicate))
   }
 
   override def fullOuterJoin(right: Table, joinPredicate: Expression): Table = {
-    joinInternal(right, Some(joinPredicate), JoinType.FULL_OUTER)
+    join(right, Some(joinPredicate), JoinType.FULL_OUTER)
   }
 
-  private def joinInternal(
+  private def join(
```

Comment:
There is a concern about whether we should do this change or not? `joinInternal`->`join`. Because if we using `join` the API is pretty clean, and if we using `joinInternal` also make scenes, due to we can not merge `joinInternal` logic into any `join` method.
do you mind do this change @dawidwys ?

Reply:
This is not part of the API. I would personally keep it separated. So personally I am in favor of keeping it as `joinInternal`.